crates/eframe/src/native/app_icon.rs

```diff
@@ -58,9 +58,8 @@ enum AppIconStatus {
 ///
 /// Since window creation can be lazy, call this every frame until it's either successfully or gave up.
 /// (See [`AppIconStatus`])
+#[profiling::function]
```

Comment:
Proc-macros like these have a tendency to increase build-times. Luckily we can opt-out of them by disabling the default features of the `profiling` crate

Reply:
That's a fair point. I don't think the `profiling` crates currently implements another abstraction that would call `puffin::profile_function!` under the hood though. So I think we have 3 options here:
1. Add `profiling::function!()` macro upstream which would call `puffin::profile_function!()`
2. Create our own `egui::profile_function!()` macro which calls `profiling::scope!()` with similar internal code to figure out the calling function name as puffin does internally in it's macro.
3. Use `profiling::scope!()` and manually type the function name.

Let me know if you have any thoughts on this. I'll have a look and see how feasible option 1 would be tonight